model/apis/axios-settings.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+const url = require('url');
+
+module.exports = (settings, body) => {
+  return Object.assign({}, settings, {
+    url: settings.uri || settings.url || url.format(settings),
+    data: settings.body || body || settings.data
+  });
+};
```

Comment:
Great work, just some thoughts

* You could use destructuring and default parameters for readability
* You could validate `settings` to check if it's an object or null

Reply:
Thanks, I have followed this suggestion